core/cu29_clock/src/lib.rs

```diff
@@ -233,6 +233,59 @@ impl From<CuTime> for OptionCuTime {
     }
 }
 
+/// Represents a time range.
+#[derive(Clone, Debug, Encode, Decode, Serialize, Deserialize)]
+pub struct CuTimeRange {
+    pub start: CuTime,
```

Comment:
`CuTime` is `Copy`, `CuTimeRange` can be copy too can't it https://doc.rust-lang.org/std/marker/trait.Copy.html#when-can-my-type-be-copy?

This would remove the need to do `clone()` in the task code

Reply:
good point, fixed.